### PR TITLE
EQL: remove version limitations for CCS

### PR DESCRIPTION
**Pull request by luigidellaquila**
Before this change, EQL queries could be executed on CCS only with all the nodes on the exact same version.

This change enables EQL CCS from an updated node to any other node that matches the [supported cross-cluster search matrix](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-supported-configurations).

Nodes of previous versions (without this fix) will still be able to run EQL CCS only to remote nodes of the exact same version

### CODE REVIEW

**Comment by luigidellaquila** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Review by astefan — COMMENTED** [review]
Before looking at the tests themselves, where again I am yet not sure they are completely needed at this point, I've added a comment regarding the versions check limitation.

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/Verifier.java [hunk 1]
```diff
@@ -151,7 +151,7 @@ Collection<Failure> verify(LogicalPlan plan, Function<String, Collection<String>
 
                 checkFilterConditionType(p, localFailures);
                 checkJoinKeyTypes(p, localFailures);
-                checkRemoteClusterOnSameVersion(p, versionIncompatibleClusters, localFailures);
```
[the remark this answers is not in the document]
Comment (luigidellaquila) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/Verifier.java:154]: I agree, if we are sure that we have other checks in place to enforce proper compatibility policies, then we can just remove this method. @bpintea anything against it?

[the remark this answers is not in the document]
Comment (bpintea) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/Verifier.java:154]: Nothing, it can now be removed.

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/qa/ccs-rolling-upgrade/src/test/java/org/elasticsearch/xpack/eql/qa/ccs_rolling_upgrade/EqlCcsRollingUpgradeIT.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                remoteIndex,
+                Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, between(1, 5)).build(),
+                "{\"properties\": {\"@timestamp\": {\"type\": \"date\"}}}",
+                null
+            );
+            int remoteNumDocs = indexDocs(remoteClient, remoteIndex, between(10, 100));
+
+            configureRemoteClusters(getNodes(remoteClient.getLowLevelClient()));
+            int iterations = between(1, 20);
+            for (int i = 0; i < iterations; i++) {
```
Comment (bpintea) [inline x-pack/plugin/eql/qa/ccs-rolling-upgrade/src/test/java/org/elasticsearch/xpack/eql/qa/ccs_rolling_upgrade/EqlCcsRollingUpgradeIT.java:184]: Curious why we went for running it multiple times.

**Review by bpintea — APPROVED** [review]
LGTM, but wondering if we shouldn't add a `checkNoBwc`, similar to [SQL](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/sql/build.gradle#L135), part of this or a follow-up PR.

File: x-pack/plugin/eql/qa/ccs-rolling-upgrade/src/test/java/org/elasticsearch/xpack/eql/qa/ccs_rolling_upgrade/EqlCcsRollingUpgradeIT.java (hunk 2 above)
Comment (luigidellaquila) [reply x-pack/plugin/eql/qa/ccs-rolling-upgrade/src/test/java/org/elasticsearch/xpack/eql/qa/ccs_rolling_upgrade/EqlCcsRollingUpgradeIT.java:184]: The dataset and the queries are randomized, this repetition just adds some more robustness to the test. Anyway, nothing critical. Also from the performance point of view it's negligible, most of the cost of this test is the cluster startup, not the test itself.

**Pull request merged**